libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -1865,24 +1870,57 @@ MQTTStatus_t MQTT_SerializeDisconnect( const MQTTFixedBuffer_t * const pBuffer )
 
 /*-----------------------------------------------------------*/
 
+MQTTStatus_t MQTT_GetPingreqPacketSize( size_t * pPacketSize )
+{
+    MQTTStatus_t status = MQTTSuccess;
+
+    if( pPacketSize == NULL )
+    {
+        LogError( "pPacketSize is NULL." );
+        status = MQTTBadParameter;
+    }
+    else
+    {
+        /* MQTT PINGREQ packets always have the same size. */
+        *pPacketSize = MQTT_PACKET_PINGREQ_SIZE;
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
 MQTTStatus_t MQTT_SerializePingreq( const MQTTFixedBuffer_t * const pBuffer )
 {
     MQTTStatus_t status = MQTTSuccess;
+    size_t pingreqPacketSize;
 
     if( pBuffer == NULL )
     {
         LogError( "pBuffer is NULL." );
         status = MQTTBadParameter;
     }
-    else if( pBuffer->size < MQTT_PACKET_PINGREQ_SIZE )
+
+    if( status == MQTTSuccess )
     {
-        LogErrorWithArgs( "Buffer size of %lu is not sufficient to hold "
-                          "serialized PINGREQ packet of size of %lu.",
-                          pBuffer->size,
-                          MQTT_PACKET_PINGREQ_SIZE );
-        status = MQTTNoMemory;
+        status = MQTT_GetPingreqPacketSize( &pingreqPacketSize );
```

Comment:
Are we taking a slightly different approach in Serialize Ping request? For other packets, the sequence that we follow is calculate the packet size and remaining length through the get packet size API and then pass it as argument to serialize API. I think it would be better to be consistent other APIs as this change is already making it closer to other APIs.